internal/log/memory_logger.go

```diff
@@ -31,21 +31,21 @@ import (
 	"go.temporal.io/sdk/log"
 )
 
-// MemoryLogger is Logger implementation that stores logs in memory (useful for testing). Use Lines() to get log lines.
-type MemoryLogger struct {
+// MemoryLoggerWithoutWith is Logger implementation that stores logs in memory (useful for testing). Use Lines() to get log lines.
```

Comment:
It would be nice to have something in the docstring that explains the strange name.

```suggestion
// MemoryLoggerWithoutWith is a Logger implementation that stores logs in memory (useful for testing). Use Lines() to get log lines.
```